lib/openai/http.rb

```diff
@@ -53,11 +53,19 @@ def to_json(string)
     # @param user_proc [Proc] The inner proc to call for each JSON object in the chunk.
     # @return [Proc] An outer proc that iterates over a raw stream, converting it to JSON.
     def to_json_stream(user_proc:)
-      proc do |chunk, _|
-        chunk.scan(/(?:data|error): (\{.*\})/i).flatten.each do |data|
-          user_proc.call(JSON.parse(data))
-        rescue JSON::ParserError
-          # Ignore invalid JSON.
+      proc do |chunk, overall_received_bytes, env|
+        Rails.logger.debug("raw chunk: #{chunk}")
+
+        if chunk.start_with?("data: {")
+          chunk.each_line do |line|
+            next unless line.start_with?("data: {")
+            line.delete_prefix!("data: ")
+            user_proc.call(JSON.parse(line))
+          end
+        elsif chunk.include?('"error": ')
```

Comment:
don't know 100% how it works, but can it just return me some JSON examples with this string?

Reply:
```
raw chunk: {
    "error": {
        "message": "The model `gpt-3.5-turbo!!!!` does not exist",
        "type": "invalid_request_error",
        "param": null,
        "code": null
    }
}
```